Sources/core/objects/OnMapBase.h

```diff
@@ -3,6 +3,7 @@
 #include <functional>
 
 #include "DrawInt.h"
+#include <SFML/System/Vector2.hpp>
```

Comment:
Do not use SFML headers at all - SFML library is used only for the sound and music processing. You should use some predefined types for vectors - VDir structure should be used for this purposes.